src/Features/Core/Portable/CodeFixes/CodeFixCollection.cs

```diff
@@ -20,19 +20,39 @@ internal class CodeFixCollection
         /// <summary>
         /// Optional fix all context, which is non-null if the given <see cref="Provider"/> supports fix all occurrences code fix.
         /// </summary>
-        public FixAllCodeActionContext FixAllContext { get; }
+        public FixAllProvider FixAllProvider { get; }
+        public FixAllContext FixAllContext { get; }
+        public IEnumerable<FixAllScope> SupportedScopes { get; }
+        public Diagnostic FirstDiagnostic { get; }
 
-        public CodeFixCollection(object provider, TextSpan span, IEnumerable<CodeFix> fixes, FixAllCodeActionContext fixAllContext = null) :
-            this(provider, span, fixes.ToImmutableArray(), fixAllContext)
+        public CodeFixCollection(
```

Comment:
The constructors differ only by the type of parameter `fixes`, one taking an ImmutableArray, and other one an IEnumerable. We can probably avoid one and instead have 2 constructors as follows:

```
public CodeFixCollection(CodeFixProvider provider, TextSpan span, ImmutableArray<CodeFix> fixes, FixAllProvider fixAllProvider, FixAllContext fixAllContext, IEnumerable<FixAllScope> supportedScopes, Diagnostic firstDiagnostic);

public CodeFixCollection(FixAllProvider provider, TextSpan span, ImmutableArray<CodeFix> fixes, Diagnostic firstDiagnostic);
```